src/Designer/backend/src/Designer/Services/Implementation/GiteaAPIWrapper/GiteaAPIWrapper.cs

```
@@ -462,10 +460,16 @@ public async Task<FileSystemObject> GetFileAsync(string org, string app, string
         /// <inheritdoc/>
         public async Task<List<FileSystemObject>> GetDirectoryAsync(string org, string app, string directoryPath, string shortCommitId)
         {
-            HttpResponseMessage response = await _httpClient.GetAsync($"repos/{org}/{app}/contents/{directoryPath}?ref={shortCommitId}");
-            return await response.Content.ReadAsAsync<List<FileSystemObject>>();
+            using HttpResponseMessage response = await _httpClient.GetAsync($"repos/{org}/{app}/contents/{directoryPath}?ref={shortCommitId}");
+            if (response.IsSuccessStatusCode)
+            {
+                return await response.Content.ReadAsAsync<List<FileSystemObject>>();
+            }
+
+            return [];
```

Should we just silently fail here?